### PR TITLE
Feat: user defined user agent

### PR DESCRIPTION
**Pull request by chaosharmonic**
What it sounds like, more or less.

I'm still figuring out how exactly the page-level bindings work, and this could maybe use some new tests, but in the meantime here's a sketch of how I'm envisioning the browser-level handling

### CODE REVIEW

**Comment by chaosharmonic** [comment]
Not *exactly* sure why this would be warning me about `fmt` on this many files when I changed two of them, but will figure that out when this is otherwise final

**Comment by chaosharmonic** [comment]
This could possibly still use a page-level `getUserAgent`, particularly for tests, but I'm not sure how that would work exactly

File: src/page.ts [hunk 1]
```diff
@@ -61,6 +61,8 @@ export type WaitForNetworkIdleOptions = {
 /** The options for sandboxing */
 export type SandboxOptions = {
   sandbox?: boolean;
+  // temporary, might not want to keep this here
+  userAgent?: string;
```
Comment (lino-levan) [inline src/page.ts:65]: Can we move this out of the sandbox options into a type that extends sandbox options? Alternatively, could you rename this to like page options or something?

File: src/browser.ts [hunk 2]
```diff
@@ -172,12 +173,15 @@ export class Browser {
     this.pages.push(page);
 
     const celestial = page.unsafelyGetCelestialBindings();
-    const { userAgent } = await celestial.Browser.getVersion();
+    const { userAgent: defaultUserAgent } = await celestial.Browser
+      .getVersion();
+
+    const userAgent = [options, this.#options]
+      .find((e) => e?.userAgent)?.userAgent ||
+      defaultUserAgent.replaceAll("Headless", "");
```
Comment (lino-levan) [inline src/browser.ts:181]: The find code here is clever but not super clear, could we make this like `options?.userAgent || this.#options?.userAgent || ...` or something?

**Review by lino-levan — CHANGES_REQUESTED** [review]

File: src/browser.ts (hunk 2 above)
Comment (chaosharmonic) [reply src/browser.ts:181]: You'll find some earlier diffs in the commit history, but TypeScript started yelling at me when I tried writing this other ways

I'll take a second look when I'm back at my laptop though

File: src/page.ts (hunk 1 above)
Comment (chaosharmonic) [reply src/page.ts:65]: Yeah, I actually forgot to even remove this comment XD

I'll rearrange this later

**Review by lino-levan — APPROVED** [review]
Looks good, thanks for the contribution!

**Pull request merged**